tests/test_utils.py

```diff
@@ -2,7 +2,7 @@
 
 
 def test_group():
-    assert list(utils.group([], 2)) == []
+    assert not list(utils.group([], 2))
```

Comment:
Please revert ALL falsey values would pass this test including None, False, 0, 0.0, {}, set(), (,).

We are asserting that the result is an empty list, not any of these other values.

Repeat below.  https://docs.astral.sh/ruff/rules/compare-to-empty-string/#known-problems